ogc-rs/src/wpad.rs

```diff
@@ -0,0 +1,129 @@
+//! The ``wpad`` module of ``ogc-rs``.
+//!
+//! This module implements a safe wrapper around the wii controller functions found in ``wpad.h``.
+
+use alloc::boxed::Box;
+use core::mem;
+
+/// Represents the wpad service. No wii remotes can be read until an instance of
+/// this struct is created. This service can only be created once!
+pub struct Wpad;
+
+/// The remote to be read for the `wpad` service.
+#[derive(Copy, Clone)]
+pub enum Remote {
+    One = 0,
+    Two = 1,
+    Three = 2,
+    Four = 3,
+}
+
+impl PartialEq<Remote> for u32 {
+    fn eq(&self, other: &Remote) -> bool {
+        *self == *other as u32
+    }
+}
+
+/// The button to be checked for the `wpad` service.
+#[derive(Copy, Clone)]
+pub enum WpadButton {
+    One = 1,
+    Two = 2,
+    B = 4,
+    A = 8,
+    Minus = 16,
+    Home = 128,
+    Left = 256,
+    Right = 512,
+    Down = 1024,
+    Up = 2048,
+    Plus = 4096,
+    Z = 65536,
+    C = 131072,
+}
```

Comment:
This does not take into account any expansion and or special controllers such as the GH3 Guitar.